### PR TITLE
8.0 add 'map_values' function (onesteinbv)

### PR DESCRIPTION
**Pull request by legalsylvain**
Port 'core' modification of openupgrade realized by OneStein BV. (branch onesteinbv:8.0-os)

### CODE REVIEW

File: openerp/openupgrade/openupgrade.py [hunk 1]
```diff
@@ -520,6 +521,57 @@ def float_to_integer(cr, table, field):
...
+
+    if write not in ('sql', 'orm'):
+        logger.exception(
+            "map_values is called with unknown value for write param: %s",
+            write)
+    if not table:
+        if not model:
+            logger.exception("map_values is called with no table and no model")
+        table = model._table
+    for old, new in mapping:
```
Comment (pedrobaeza) [inline openerp/openupgrade/openupgrade.py:560]: You should also check if source_column is different from target_column.

Comment (legalsylvain) [inline openerp/openupgrade/openupgrade.py:560]: Fixed. Are you ok with the changes ?

**Comment by StefanRijnhart** [comment]
:-1: Taken from https://github.com/OpenUpgrade/OpenUpgrade/pull/80/files. Review this first please.

**Comment by legalsylvain** [comment]
@StefanRijnhart : Agree. 
:-1: 
Thanks.

**Pull request closed without merging**